### DISCUSSION TITLE
Export to AMPL

### DISCUSSION THREAD

**Discussion by zoharl3** [yalmip/YALMIP#1162] [General discussion] [answered]
I'd like to export my model into AMPL in order to test it with the different optimization packages on NEOS, as suggested here:

https://or.stackexchange.com/questions/8822/methods-for-binary-linear-programming

I tried saveampl(). It works fine for small problems, but it didn't finish exporting a large problem, and I terminated it after hours. It seems to process each constraint manually.

I wonder why you didn't simply extracted the problem matrices and saved them as AMPL .dat instead, which is what I'm writing right now.

---

Johan, a few days ago I posted here a question for you (on a math forum, related to the link above), as an optimization expert, to get your opinion on a specific problem so you might refer me to somewhere to read. The question has disappeared, and I'm not sure if it's a gentle hint not to post here optimization-related questions that aren't directly related to yalmip(?).

**Comment by johanlofberg** [comment, marked as answer]
I never really cared about the ampl/gams/cplexlp/... exports so super low priority and didn't even know that they have a numerical format. Do you have a good description of it somewhere?

I removed that post as it was only a hint that I should check it, which I did, but after that I don't see any value for other for it to remain here as this is primarily YALMIP.

**Reply by zoharl3** [reply]
(posted it previously as an answer by misake)

I never cared for them either, but you must admit that it's not a bad idea to have a common format, at least for comparing packages conveniently (unless you already have all of them installed).

Right now, I'm trying to export to .dat using

https://au.mathworks.com/matlabcentral/fileexchange/39795-ampl-data-file-toolbox

Here there's an example how to use it:

https://www.math.wsu.edu/faculty/bkrishna/FilesMath567/S17/Handouts/Handout_AMPL_1.pdf

First issue: the toolbox only handles full matrices. 

I'll try to update when I figure something out.

---

Thanks for the clarification about my unrelated question and for taking a look. I'm assuming that you won't mind such questions from time to time, and you'll simply delete them if you have nothing to say.

**Comment by zoharl3** [comment]
My initial impression is that modeling languages such as AMPL, GAMS, and pyomo are completely useless: they don't seem to support linear algebra.  I merely wanted to solve a canonical LP

min c'*x
s.t. A*x <= b

(two lines in yalmip.)

Ridiculous.
Seems like my initial judgment of your implementation was naïve.

I posted some questions 

https://or.stackexchange.com/questions/8871/compare-optimization-packages-on-a-specific-problem

Maybe there's a workaround.

**Reply by zoharl3** [reply]
GAMS seems to have a better matlab support.

I didn't try it, but the constraint Ax <= b could be formulated in GAMS with a one line:

`EquationName(i).. sum(j, A(i,j)*x(j)) =l= b(i);`

Probably the other suites have something similar.
